src/main/java/nl/dtls/fairdatapoint/service/impl/PurlPIDSystemImpl.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * The MIT License
+ * Copyright © 2017 DTL
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package nl.dtls.fairdatapoint.service.impl;
+
+import com.google.common.base.Preconditions;
+import nl.dtl.fairmetadata4j.model.FDPMetadata;
+import nl.dtl.fairmetadata4j.model.Metadata;
+import nl.dtls.fairdatapoint.service.PIDSystem;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+/**
+ * Implementation of purl.org PID system
+ *
+ * @author Rajaram Kaliyaperumal <rr.kaliyaperumal@gmail.com>
+ * @author Kees Burger <kees.burger@dtls.nl>
+ * @since 2018-06-06
+ * @version 0.1
+ */
+public class PurlPIDSystemImpl implements PIDSystem {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PurlPIDSystemImpl.class);
+    private static final ValueFactory VALUEFACTORY = SimpleValueFactory.getInstance();
+    private static String fdpUri = "";
```

Comment:
Why is `fdpUri` static?

Reply:
Or maybe a more fundamental question: why is `fdpUri` a class member (and not a method scoped variable)? What happens if `getURI` is invoked with a catalog metadata instance?

Reply:
If the makes a catalog POST call on a empty FDP with no repository metadata then the controller will hold this POST call, it will first create the repository metadata and then creates catalog metadata.

Reply:
The other option is we should move this to the config file. Let the fdp manager to set the fdp base URI